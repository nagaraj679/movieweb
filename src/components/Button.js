import React from 'react';
import PropTypes from 'prop-types';

// class Button extends React.Component {
//   render() {
//     return (
//       <button className="button" onClick={() => { this.props.handleClick() }}>{this.props.children}</button>
//     )
//   }
// }

//I can use a functional component here because I just need to pass props,
//I don't need life cycle method and state
//This is a Pure Components because it is declared as a function that has no state and returns the same markup given the same props
const  Button = props => {
  return <button className="button" onClick={() => { props.handleClick() }}>{props.children}</button>;
}

Button.propTypes = {
  handleClick: PropTypes.func,
}

export default Button;
