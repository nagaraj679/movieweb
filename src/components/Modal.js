import React from 'react';
///import PropTypes from 'prop-types';
import Button from './Button';
import ReactDOM from  'react-dom';
import connect from 'react-redux/es/connect/connect';
import { seats, transactionModal} from '../actions';

class Modal extends React.Component {
  constructor() {
    super();
  }

  render() {
    // Render nothing if the msg prop is empty
    if (!this.props.activeTransactionModal) {
      return null; 
    }
    //Use of createPortal to render the modal inside the document body (to prevent z-index problems)
    return (
      ReactDOM.createPortal(
        <div className='bg-modal' onClick={this.props.onCloseModal}>
          <div className='modal-content'>
            <p>{this.props.activeTransactionModal}</p>
            <Button handleClick={this.props.onCloseModal}>
              Close
            </Button>
          </div>
        </div>,
        document.getElementById('modal-root') )
    );
  }
}

const mapStateToProps = (state) => {
  return {
    activeTransactionModal: state.activeTransactionModal,
  }
};

function mapDispatchToProps(dispatch) {
  return {
    onCloseModal : (data) => {
      dispatch(seats(null));
      dispatch(transactionModal(null));
    },
  };
}

//Connects the Modal component to the Redux store.
export default connect(mapStateToProps, mapDispatchToProps)(Modal);


