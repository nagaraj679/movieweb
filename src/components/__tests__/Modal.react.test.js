import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Modal from '../Modal';
import Button from '../Button';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

describe('Modal', () => {

    // add a div with #modal-root id to the global body
    const modalRoot = global.document.createElement('div');
    modalRoot.setAttribute('id', 'modal-root');
    const body = global.document.querySelector('body');
    body.appendChild(modalRoot);

    it('should be defined', () => {
      expect(Modal).toBeDefined();
    });

    it('should render correctly', () => {
      const tree = shallow(
        <Modal msg={"Success!"} onCloseModal={()=>{}} />
      );
      expect(tree).toMatchSnapshot();
    });

    it('should not render', () => {
      const tree = shallow(
        <Modal msg={""} onCloseModal={()=>{}} />
      );
      expect(tree.find('div')).toHaveLength(0);
    });

    // const mockFn = jest.fn();
    // it('should contain a button', () => {
    //   const tree = shallow(
    //     <Modal msg={"hey!"} onCloseModal={mockFn} />
    //   );
    //   expect(tree.find(Button)).toHaveLength(1);
    // })

});