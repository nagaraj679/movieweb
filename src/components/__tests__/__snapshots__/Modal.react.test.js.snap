// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    msg="Success!"
    onCloseModal={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "$$typeof": Symbol(react.portal),
    "children": <div
      className="bg-modal"
    >
      <div
        className="modal-content"
      >
        <p>
          Success!
        </p>
        <Button
          handleClick={[Function]}
        >
          Close
        </Button>
      </div>
    </div>,
    "containerInfo": <div
      id="modal-root"
    />,
    "implementation": null,
    "key": null,
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "$$typeof": Symbol(react.portal),
      "children": <div
        className="bg-modal"
      >
        <div
          className="modal-content"
        >
          <p>
            Success!
          </p>
          <Button
            handleClick={[Function]}
          >
            Close
          </Button>
        </div>
      </div>,
      "containerInfo": <div
        id="modal-root"
      />,
      "implementation": null,
      "key": null,
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
